types/types.go

```diff
@@ -53,22 +54,30 @@ type AlertStatus struct {
 // Marker helps to mark alerts as silenced and/or inhibited.
 // All methods are goroutine-safe.
 type Marker interface {
-	// SetActiveOrSilenced replaces the previous SilencedBy by the provided IDs of
-	// active and pending silences, including the version number of the
-	// silences state. The set of provided IDs is supposed to represent the
-	// complete set of relevant silences. If no active silence IDs are provided and
-	// InhibitedBy is already empty, it sets the provided alert to AlertStateActive.
-	// Otherwise, it sets the provided alert to AlertStateSuppressed.
+	// SetActiveOrSilenced replaces the previous SilencedBy by the provided
+	// IDs of active and pending silences, including the version number of
+	// the silences state. The set of provided IDs is supposed to represent
+	// the complete set of relevant silences. If no active silence IDs are
+	// provided and both InhibitedBy and MutedBy are empty, it sets the
+	// provided alert to AlertStateActive. Otherwise, it sets the provided
+	// alert to AlertStateSuppressed.
 	SetActiveOrSilenced(alert model.Fingerprint, version int, activeSilenceIDs, pendingSilenceIDs []string)
-	// SetInhibited replaces the previous InhibitedBy by the provided IDs of
-	// alerts. In contrast to SetActiveOrSilenced, the set of provided IDs is not
-	// expected to represent the complete set of inhibiting alerts. (In
-	// practice, this method is only called with one or zero IDs. However,
+	// SetInhibited replaces the previous InhibitedBy by the provided IDs
+	// of alerts. In contrast to SetActiveOrSilenced, the set of provided
+	// IDs is not expected to represent the complete set of inhibiting alerts.
+	// In practice, this method is only called with one or zero IDs. However,
 	// this expectation might change in the future. If no IDs are provided
-	// and InhibitedBy is already empty, it sets the provided alert to
+	// and InhibitedBy and SilencedBy are empty, it sets the provided alert to
 	// AlertStateActive. Otherwise, it sets the provided alert to
 	// AlertStateSuppressed.
 	SetInhibited(alert model.Fingerprint, alertIDs ...string)
+	// SetMuted replaces the previous MutedBy by the provided time intervals.
+	// Unlike SetInhibited, the set of provided time intervals is expected
+	// to represent the complete set of relevant time intervals. If no time
+	// intervals are provided and both InhibitedBy and SilencedBy are empty,
+	// it sets the provided alert to AlertStateActive. Otherwise, it sets the
+	// provided alert to AlertStateSuppressed.
+	SetMuted(alert model.Fingerprint, alertIDs ...string)
```

Comment:
```suggestion
	// SetMuted replaces the previous MutedBy by the provided time intervals.
	// Similarly to SetInhibited and in contrast to SetActiveOrSilenced, the set of provided time intervals is expected
	// to not represent the complete set of relevant time intervals. If no time
	// intervals are provided and both InhibitedBy and SilencedBy are empty,
	// it sets the provided alert to AlertStateActive. Otherwise, it sets the
	// provided alert to AlertStateSuppressed.
	SetMuted(alert model.Fingerprint, alertIDs ...string)
```

Reply:
My understanding of both `SetMuted` and `SetInhibited` is that if you have an alert that was previously muted by a rule or interval if that alert gets muted again it'll override the previous one - this can't happen in `SetActiveOrSilenced` because we'll query that's affecting.

Reply:
Ah no, that's not how it works in the PR. I'm recording the names of **all** time intervals that muted it, not just the first time interval. I understand that silences work the same?

Reply:
Ah. But if there the alert matches two routes, and both those routes have different mute time intervals, just one of them will be recorded. :frowning_face: